.github/workflows/labeler.yml

```diff
@@ -53,12 +53,14 @@ jobs:
         env:
           event_json: "${{ toJSON(github.event) }}"
           GITHUB_TOKEN: ${{ steps.create_token.outputs.token }}
-        run:
-          ./venv/bin/python hacking/pr_labeler/label.py issue ${{ github.event.issue.number || inputs.number }}
+          number: "${{ github.event.issue.number || inputs.number }}"
+        run: |
+          ./venv/bin/python hacking/pr_labeler/label.py issue "${number}"
```

Comment:
Moving things to env vars seems silly. The real fix is quoting it, and the env var is just misleading, it's not a real fix.

Although, the quoted value is as vulnerable to command injections as the original version.

OTOH, the number is coming from the GH event (which we trust) or from the manual workflow trigger, which is exploitable by people who already have access to the repository anyway. You're basically protecting yourself from you. This doesn't really make sense to me, I don't see any legitimate attack vector here.
The only realistic attack surface I recognize is the `label.py` script, which would only be dangerous if somebody were to smuggle changes to it (or its deps) by getting a patch merged to the repo, which effectively means that any changes to the script/deps should be reviewed with extra caution. I think, this is one of the cases when we'd want to use the codeowners config and require reviews from owners, maybe.

Reply:
> Although, the quoted value is as vulnerable to command injections as the original version.

No, not really. With this change, the env var `number` is set to that exact text of `${{ github.event.issue.number || inputs.number }}` and then passed verbatim as `argv[2]`. When setting `inputs.number` to `$(unsafe-command-here)`, `$(unsafe-command-here)` will no longer be expanded after this change.

> OTOH, the number is coming from the GH event (which we trust) or from the manual workflow trigger, which is exploitable by people who already have access to the repository anyway.

Right, I said as much in the PR description. In any case, I think this makes the script more readable, and it's a good idea to stick to this practice.

Reply:
Fair, marking this thread as resolved.